test/unit/test_expr.ml

```diff
@@ -614,13 +614,24 @@ let test_simplify_concat_i32_symbol _ =
   let concated = Expr.concat d (Expr.concat c (Expr.concat b a)) in
   check concated x
 
+let test_fp_nan_not_geffects _ =
+  let open Infix in
+  let ty = Ty.Ty_fp 32 in
+  let x = symbol "x" ty in
+  let y = symbol "y" ty in
+  (* x != x makes isNaN(x) *)
+  let expr = Expr.unop ty Not (Expr.relop ty Ge x y) in
+  let expected = Expr.unop ty Not (Expr.relop ty Le y x) in
```

Comment:
```suggestion
  let expr = Expr.unop Ty_bool Not (Expr.relop ty Ge x y) in
  let expected = Expr.unop Ty_bool Not (Expr.relop ty Le y x) in
```

The `Not` should be encoded as the boolean operator of negation. Other than that every thing looks good to go!